lib/metasploit/framework/login_scanner/x3.rb

```diff
@@ -0,0 +1,115 @@
+# -*- coding: binary -*-
+
+# require 'msf/core'
+# require 'msf/core/exploit/tcp'
+require 'metasploit/framework/login_scanner/base'
+require 'metasploit/framework/login_scanner/rex_socket'
+require 'metasploit/framework/tcp/client'
+
+module Metasploit
+  module Framework
+    module LoginScanner
+      class X3
+        include Metasploit::Framework::LoginScanner::Base
+        include Metasploit::Framework::LoginScanner::RexSocket
+        include Metasploit::Framework::Tcp::Client
+
+        DEFAULT_PORT = 50000
+        REALM_KEY = nil
+
+        def encrypt_pass(inp)
+          # check if it's already encrypted
+          if inp.include?('CRYPT:')
+            return inp
+          end
+
+          num2 = inp.length
+          num = 17
+          ret = ''
+          charset_0 = 'cromanwqxfzpgedkvstjhyilu'.split('')
+          xyz = 'zxWyZxzvwYzxZXxxZWWyWxYXz'.split('')
+          charset_1 = 'cf2tln3yuVkDr7oPaQ8bsSd4x'.split('')
```

Comment:
Instead of splitting on an empty string you can use the `String#chars` method.

```suggestion
          charset_0 = 'cromanwqxfzpgedkvstjhyilu'.chars
          xyz = 'zxWyZxzvwYzxZXxxZWWyWxYXz'.chars
          charset_1 = 'cf2tln3yuVkDr7oPaQ8bsSd4x'.chars
```